tflint/tflint.go

```diff
@@ -0,0 +1,117 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package tflint
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/gruntwork-io/terragrunt/config"
+	"github.com/gruntwork-io/terragrunt/errors"
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terragrunt/util"
+	"github.com/terraform-linters/tflint/cmd"
+)
+
+func RunTflintWithOpts(terragruntOptions *options.TerragruntOptions, terragruntConfig *config.TerragruntConfig) error {
+	configFile, err := findTflintConfigInProject(terragruntOptions)
+	if err != nil {
+		return err
+	}
+	terragruntOptions.Logger.Debugf("Found .tflint.hcl file in %s", configFile)
+
+	variables, err := inputsToTflintVar(terragruntConfig.Inputs)
+	if err != nil {
+		return err
+	}
+
+	terragruntOptions.Logger.Debugf("Initializing tflint in directory %s", terragruntOptions.WorkingDir)
+	cli := cmd.NewCLI(terragruntOptions.Writer, terragruntOptions.ErrWriter)
+
+	initArgs := []string{"tflint", "--init", "--config", configFile, terragruntOptions.WorkingDir}
+	statusCode := cli.Run(initArgs)
+	if statusCode != 0 {
+		return errors.WithStackTrace(ErrorRunningTflint{args: initArgs})
+	}
+
+	args := []string{"tflint"}
+	args = append(args, variables...)
+	args = append(args, "--config", configFile)
+	args = append(args, "--module")
+	args = append(args, terragruntOptions.WorkingDir)
+
+	terragruntOptions.Logger.Debugf("Running tflint with args %v", args)
+	statusCode = cli.Run(args)
+
+	if statusCode == cmd.ExitCodeError {
+		return errors.WithStackTrace(ErrorRunningTflint{args: initArgs})
+	} else if statusCode == cmd.ExitCodeIssuesFound {
+		// When tflint finds issues in the project, it prints out but does not stop execution of Terragrunt.
+		terragruntOptions.Logger.Warnf("Tflint found issues in the project. Check for the tflint logs.")
+	} else if statusCode == cmd.ExitCodeOK {
+		terragruntOptions.Logger.Info("Tflint has run successfully. No issues found.")
+	} else {
+		terragruntOptions.Logger.Warnf("Unknown status code from tflint: %d", statusCode)
+	}
+
+	return nil
+}
+
+func inputsToTflintVar(inputs map[string]interface{}) ([]string, error) {
+	var variables []string
+	for key, value := range inputs {
+		varValue, err := util.AsTerraformEnvVarJsonValue(value)
+		if err != nil {
+			return nil, err
+		}
+
+		newVar := fmt.Sprintf("--var=%s=%s", key, varValue)
+		variables = append(variables, newVar)
+	}
+	return variables, nil
+}
+
+// findTflintConfigInProjects looks for a .tflint.hcl file in the current folder or it's parents.
+func findTflintConfigInProject(terragruntOptions *options.TerragruntOptions) (string, error) {
+	previousDir := terragruntOptions.WorkingDir
+
+	// To avoid getting into an accidental infinite loop (e.g. do to cyclical symlinks), set a max on the number of
+	// parent folders we'll check
+	for i := 0; i < terragruntOptions.MaxFoldersToCheck; i++ {
```

Comment:
NIT: seems like a common "find in parent folders" method could be extracted from this code, rather than copy pasted.